python/proton/_message.py

```diff
@@ -517,16 +524,20 @@ def encode(self):
         self._pre_encode()
         sz = 16
         while True:
-            err, data = pn_message_encode(self._msg, sz)
-            if err == PN_OVERFLOW:
+            size = ffi.new('size_t *', sz)
+            bytes = ffi.new("char []", sz)
+            status_code = pn_message_encode(self._msg, bytes, size)
+            if status_code == PN_OVERFLOW:
                 sz *= 2
                 continue
             else:
-                self._check(err)
-                return data
+                self._check(status_code)
+                return ffi.buffer(bytes, size[0])[:]
```

Comment:
What does `[:]` do here? Create a copy? Convert to `bytes`?

Reply:
it returns bytes